rfcs/999-shipperctl-admin.adoc

```diff
@@ -0,0 +1,94 @@
+= 999 Introduce _shipperctl_ command line utility
+Parham Doustdar <parham.doustdar@booking.com>
+:RFC-Status: Draft
+
+== Abstract
+
+Now that we're working to simplify setup, administration, and scripting, I think it makes sense to re-open the discussion around a _shipperctl_ CLI for Shipper. The most valuable functionality we can get out of such a tool now is *simplifying Shipper's setup*, so this RFC will focus only on the `+shipperctl admin clusters apply`+ command.
+
+== Motivation
+
+The current process of configuring a Shipper installation for production is quite complicated and error prone: Shipper CRDs, service accounts, role bindings, and cluster secrets must be created, only to name a few of the actions required to properly set it up.
+
+Having a complicated process also complicates other parts of our project, specifically Shipper documentation and quick-start guides. In addition to documentation and guides, it also makes it very difficult for third parties interested in Shipper to properly evaluate it. The team strongly believes that having a simple administration tool that performs the majority of those error prone and complicated procedures will increase the chances of such third parties to consider and use Shipper in their environments.
+
+== Reference level explanation
+
+`+shipperctl admin clusters apply`+ performs a bulk operation setting up all clusters contained in a Shipper cluster configuration file. This file contains all clusters to be configured for Shipper operation, for both Management and Application clusters.
+
+.Cluster configuration file example
+[source,yaml]
+----
+managementClusters:
+- name:    eu-management     # <1>
+- name:    us-management
+  context: gke-us-management # <2>
+
+applicationClusters:
+- name:   eu-application     # <1>
+  region: eu-west1
+  zone:   a
+  capabilities:
+    - gdpr
+    - ipv4
+    - kubernetes1.12
+- name:    us-application
+  context: gke-us-application # <2>
+  region:  us-west1
+  zone:    a
+  capabilities:
+    - ipv4
+    - ssd-local-disk
+----
+<1> Without the `context` field present, `name` is considered a valid and existing _kubectl_ context, and will be used both to gather connection details to perform relevant operations on the cluster to set up Shipper and to name cluster specific manifests in the Management Cluster.
+<2> With the `context` field present, the `name` value is used only to name cluster specific manifests in the Management Cluster. Connection details are gathered using the specified `context` value.
+
+.*shipperctl admin clusters apply* arguments
+|===
+|Option|Description
+
+|`-f, --file`
+| Shipper cluster configuration file to be applied
+
+|`--dry-run`
+| Print out what actions it would do, according to the given cluster configuration file
+
+|`-n, --namespace`
+| Shipper system namespace to be created in all clusters
+
+|`-k, --kubeconfig`
```

Comment:
I'd avoid using `-k`, because it isn't a shortcut that `kubectl` uses for `kubeconfig`.